src/lightning/pytorch/trainer/configuration_validator.py

```diff
@@ -98,11 +88,25 @@ def __verify_train_val_loop_configuration(trainer: "pl.Trainer", model: "pl.Ligh
             category=PossibleUserWarning,
         )
 
+    # check legacy hooks are not present
+    if callable(getattr(model, "training_epoch_end", None)):
```

Comment:
cc @edenafek for a review

Reply:
Maybe "Support for training_epoch_end has been removed in v2.0.0, use the on_train_epoch_end hook instead. To access outputs, save them in-memory an instance attributes."